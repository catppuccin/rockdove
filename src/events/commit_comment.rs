use octocrab::models::webhook_events::{WebhookEvent, payload::CommitCommentWebhookEventPayload};

use crate::{
    colors::COMMIT_COLOR,
    embed_builder::EmbedBuilder,
    errors::{RockdoveError, RockdoveResult},
};

pub fn make_embed(
    event: WebhookEvent,
    specifics: &CommitCommentWebhookEventPayload,
) -> RockdoveResult<Option<EmbedBuilder>> {
    let repo = event
        .repository
        .ok_or_else(|| RockdoveError::MissingField {
            event_type: event.kind.clone(),
            field: "repository",
        })?;

    let mut embed = EmbedBuilder::default();

    let repo_name = repo.full_name.unwrap_or(repo.name);

    embed.title(&format!(
        "[{}] New comment on commit {}",
        repo_name, specifics.comment.commit_id,
    ));
    embed.url(specifics.comment.html_url.as_str());
    embed.description(
        specifics
            .comment
            .body
            .as_ref()
            .ok_or_else(|| RockdoveError::MissingField {
                event_type: event.kind.clone(),
                field: "comment.body",
            })?
            .as_str(),
    );
    embed.color(COMMIT_COLOR);

    Ok(Some(embed))
}

#[cfg(test)]
mod tests {
    use crate::snapshot_test;

    #[test]
    fn created() {
        snapshot_test!("commit_comment", "created");
    }
}
