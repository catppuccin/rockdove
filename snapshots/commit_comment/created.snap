---
source: src/events/commit_comment.rs
info:
  author_name_length: 8
  colour_hex: "#94E2D5"
  description_length: 20
  title_length: 87
---
embeds:
  - author:
      icon_url: "https://avatars.githubusercontent.com/u/58985301?v=4"
      name: sgoudham
      url: "https://github.com/sgoudham"
    color: 9757397
    description: testing for rockdove
    title: "[catppuccin-rfc/polybar] New comment on commit 8c687cd0fc28fa88bcfcab9932f885a829726b3d"
    url: "https://github.com/catppuccin-rfc/polybar/commit/8c687cd0fc28fa88bcfcab9932f885a829726b3d#commitcomment-147026704"
